Assets/Scripts/DrawDetector.cs

```diff
@@ -1 +1,20 @@
+using UnityEngine;
 
+public class DrawDetector : MonoBehaviour
```

Comment:
There is no reason for this class to be monobehaviour it does not use any properties of an object it is attached to.
Please remove monobeh and just pass needed dependencies from another class which has it.